commit e9186663b868668a5fecfa4c7d2238fb9326152e (refs/changes/35/2935/1)
Author: Jerry Jelinek <jerry.jelinek@joyent.com>
Date:   2017-11-10T20:24:21+00:00 (1 year, 11 months ago)
    
    OS-6407 lx: move mmap and mremap support in-kernel [fix DEBUG build]

diff --git a/usr/src/uts/common/brand/lx/syscall/lx_mem.c b/usr/src/uts/common/brand/lx/syscall/lx_mem.c
index cb4cc4ce5e..836edbfa14 100644
--- a/usr/src/uts/common/brand/lx/syscall/lx_mem.c
+++ b/usr/src/uts/common/brand/lx/syscall/lx_mem.c
@@ -1047,10 +1047,10 @@ lx_u2u_copy(void *src, void *dst, size_t len)
 	struct as *p_as = curproc->p_as;
 
 	/* Both sides should be page aligned since they're from smmap64 */
-	ASSERT((src & PAGEOFFSET) == 0);
-	ASSERT((dst & PAGEOFFSET) == 0);
+	ASSERT(((uintptr_t)src & PAGEOFFSET) == 0);
+	ASSERT(((uintptr_t)dst & PAGEOFFSET) == 0);
 	/* Both came from mmap, so they should be valid user pointers */
-	ASSERT(src < USERLIMIT && dst < USERLIMIT);
+	ASSERT((uintptr_t)src < USERLIMIT && (uintptr_t)dst < USERLIMIT);
 
 	sp = src;
 	dp = dst;
